### PR TITLE
Add java solution for CSES Four Sum

### PR DESCRIPTION
**Pull request by andreionoie**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/gold/cses-1642.mdx [hunk 1]
```diff
@@ -76,5 +76,37 @@ int main() {
...
 </CPPSection>
+<JavaSection>
+```java
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class Main {
+	public static void main(String[] args) {
+		final Scanner s = new Scanner(System.in);
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1642.mdx:87]: use `BufferedReader` or `Kattio` for consistency

File: solutions/gold/cses-1642.mdx [hunk 2]
```diff
@@ -76,5 +76,37 @@ int main() {
...
+		final int n = s.nextInt();
+		final int x = s.nextInt();
+		final int[] arr = new int[n + 1];  // 1-indexed array for this problem
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1642.mdx:90]: `final` not needed

File: solutions/gold/cses-1642.mdx [hunk 3]
```diff
@@ -76,5 +76,37 @@ int main() {
...
+		for (int i = 1; i <= n; i++) arr[i] = s.nextInt();
+
+		final Map<Integer, String> twoSum = new HashMap<>();
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1642.mdx:93]: yeah

File: solutions/gold/cses-1642.mdx [hunk 4]
```diff
@@ -76,5 +76,37 @@ int main() {
...
+		for (int i = 1; i <= n; i++) {
+			for (int j = i + 1; j <= n; j++) {
+				final int differenceToX = x - arr[i] - arr[j];
+				if (twoSum.containsKey(differenceToX)) {
+					System.out.println(twoSum.get(differenceToX) + " " + i +
+					                   " " + j);
```
Comment (IDON-TEXIST) [inline solutions/gold/cses-1642.mdx:99]: This looks worse than just going a couple characters over 80.

Comment (SansPapyrus683) [inline solutions/gold/cses-1642.mdx:99]: the line wrapping is ugly- put the result in a variable first, perhaps?

File: solutions/gold/cses-1642.mdx [hunk 5]
```diff
@@ -76,5 +76,37 @@ int main() {
...
+					return;
+				}
+			}
+			for (int j = i - 1; j >= 1; j--)
+				twoSum.put(arr[i] + arr[j], i + " " + j);
```
Comment (SansPapyrus683) [inline solutions/gold/cses-1642.mdx:104]: braces

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by IDON-TEXIST — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Review by IDON-TEXIST — APPROVED** [review]

**Pull request merged**